traffic_prophet/countmatch/reader.py

```diff
@@ -30,62 +29,45 @@ def __init__(self, count_id, centreline_id, direction, year,
         super().__init__(count_id, centreline_id, direction, data,
                          is_permanent=is_permanent)
 
-    @staticmethod
-    def regularize_timeseries(rd):
-        """Regularize count data.
-
-        Copy's the object's count data and averages out observations with
-        duplicate timestamps.
+    @classmethod
+    def from_raw_data(cls, rd):
+        """Processes input data.
 
         Parameters
         ----------
-        rd : RawCount
-            Raw count data object.
-
+        rd : dict
+            Raw data, with entries for 'centreline_id', 'direction',
+            'year' and 'data'.  The first three are integers, while 'data'
+            is a `pandas.DataFrame` with 'Count' column and either 'Timestamp'
+            or 'Date' column.
         """
+        if 'Date' not in rd['data'].columns:
```

Comment:
I don't see why this is important any longer - `RawAnnualCount` would otherwise accept a duck-type `rd`, and would fail if subsequent steps require a `Date` column, which would be more Pythonic.